nano/node/common.cpp

```diff
@@ -51,9 +42,6 @@ uint64_t nano::ip_address_hash_raw (boost::asio::ip::address const & ip_a, uint1
 
 nano::message_header::message_header (nano::message_type type_a) :
 	network (nano::network_constants::active_network),
-	version_max (get_protocol_constants ().protocol_version),
```

Comment:
The goal with leaving them uninitialized is that a compiler tool could detect uninitialized usages. Alternatively we could initialized them with a value like 0xff and then debug_assert that they are not so when serializing.

Reply:
It's just that it is another exception that we have to remember - that these fields are not always set.
And it is a random exception. For example, why set version only bytes in the network send function? Why not set the magic and the networks ID byte there too? In fact, I wonder if it would be possible to set the whole message header there.